### PR TITLE
WIP: OpenGL Example

### PR DESCRIPTION
**Pull request by sandhawalia**
Addresses #310 

### Build

```
docker-compose -f docker/docker-compose.yml build --build-arg GEN_OPENGL_EXT=1 vpf
```

### Run 

```
python SampleOpenGL.py --help
usage: This sample decodes input video to OpenGL Texture.
Requires the GL Utility Toolkit (GLUT) and pyCUDA compiled with GL support
Controls: c -> toggles cpu path (CUDA->cpu->OpenGL)
          g -> toggles gpu path (CUDA->OpenGL)
          q -> exit demo.
       [-h] -g GPU_ID -e ENCODED_FILE_PATH [-v]

optional arguments:
  -h, --help            show this help message and exit
  -g GPU_ID, --gpu-id GPU_ID
                        GPU id, check nvidia-smi
  -e ENCODED_FILE_PATH, --encoded-file-path ENCODED_FILE_PATH
                        Encoded video file (read from)
  -v, --verbose         Verbose
```

### CODE REVIEW

File: SampleOpenGL.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
+#
+# Copyright 2019 NVIDIA Corporation
```
Comment (rarzumanyan) [inline SampleOpenGL.py:2]: Please fix the copyright header.

File: SampleOpenGL.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
+#
+# Copyright 2022 NVIDIA Corporation
```
Comment (rarzumanyan) [inline SampleOpenGL.py:2]: Copyright shall belong to the author ;)
This sample wasn't developed by NVIDIA.

Comment (sandhawalia) [reply SampleOpenGL.py:2]: Ah!

Comment (rarzumanyan) [reply SampleOpenGL.py:2]: @Yves33 and @sandhawalia 

Please add your copyright(s) here.

Comment (rarzumanyan) [reply SampleOpenGL.py:2]: @Yves33 and @sandhawalia 

Please add your copyright(s) here.

**Comment by Yves33** [comment]
In you code, pycuda.gl.RegisterImage is not necessary as the copy between cuda and gl is done through the buffer. The texture is not shared.

**Comment by rarzumanyan** [comment]
@Yves33, @sandhawalia 
Thanks a lot for your contribution!

Changing target branch to `v_1.1.2` for testing, merging.
Will merge with `master` a bit later after I check it locally.

**Pull request merged**